### ISSUE TITLE
hs.notify : can we get a callback when the user clicks cancel?

### ISSUE THREAD

**Issue by alice0meta** [Hammerspoon/hammerspoon#1574]
if the user has set Hammerspoon notifications to Alert in the Notification Center pane of System Preferences, then the notification has two buttons (the action button and the other button). We can then get a callback only if the user clicks the content or the action button - the other button is a hardwired cancel button. Can we get a callback when the user clicks the other button?

**Comment by asmagill** [comment]
I don't believe so; cancel causes the notification center to clear the notification and the Hammerspoon delegate is never invoked.

If you know of an application or code that does capture the "cancel" action, I can take a look at it, but so far I don't see anything that looks promising.

**Comment by alice0meta** [comment]
Is it possible to at poll for it?

**Comment by asmagill** [comment]
ATM, no... however, looking at `NSUserNotificationCenter`'s `deliveredNotifications` property may provide a way for us to check and see if the userdata is still in the list.

It may be few days before I can get to this, so if someone wants to take a crack at it sooner, I won't object.

**Comment by latenitefilms** [comment]
This would be awesome if possible!! Thanks @asmagill !

**Comment by asmagill** [comment]
@latenitefilms, @alice0meta can either of you confirm if pull #1578, which has been merged into the master branch of Hammerspoon now, provides you with what you need to poll for this?

**Comment by latenitefilms** [comment]
@asmagill - I've been meaning to test this out. Will try and get to it ASAP.

**Comment by latenitefilms** [comment]
@cmsj & @asmagill - This issue can now be closed thanks to #1578
